application/console/migrations/m250318_201324_add_mfa_type_recovery.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+use yii\db\Migration;
+use common\models\Mfa;
+
+/**
+ * Class m250318_201324_add_mfa_type_recovery
+ */
+class m250318_201324_add_mfa_type_recovery extends Migration
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function safeUp()
+    {
+
+        // add recovery mfa type
+        $this->alterColumn(
+            '{{mfa}}',
+            'type',
+            "enum('totp','u2f','backupcode','manager','webauthn', 'recovery') not null"
+        );
+
+        //Adding in Recovery MFA emails
+        $this->alterColumn(
+            '{{email_log}}',
+            'message_type',
+            "enum('invite','welcome','mfa-rate-limit','password-changed','get-backup-codes','refresh-backup-codes','lost-security-key','mfa-option-added','mfa-option-removed','mfa-enabled','mfa-disabled','method-verify','mfa-manager','mfa-manager-help','method-reminder','method-purged','password-expiring','password-expired','password-pwned','ext-group-sync-errors','abandoned-users', 'mfa-recovery', 'mfa-recovery-help') null"
+        );
+
+        //Updating all manager email records to recovery emails
+        $this->update("{{email_log}}", ["message_type" => "mfa-recovery"], ["message_type" => "mfa-manager"]);
+
+        $this->update("{{email_log}}", ["message_type" => "mfa-recovery-help"], ["message_type" => "mfa-manager-help"]);
```

Comment:
I hesitate to modify history records, though I think what you did here will work. I just wonder if it's better to leave those records untouched.

Actually, perhaps your approach is better, since going forward we will refer to these as mfa-recovery(-help) emails, and the EmailLog is for tracking when/whether we have already sent such an email. Leaving the old records as "manager" could result in another email being sent (since the EmailLog won't have a record that a "recovery" email was sent).

Okay, my rambling here is done. No need to change your approach here after all.